.eslintrc.cjs

```diff
@@ -1,63 +1,139 @@
+/**
+ * @type {ESLint.ConfigData}
+ */
 module.exports = {
-  env: {
-    commonjs: true,
-    es2022: true,
-    node: true,
-    jest: true,
-    'jest/globals': true
-  },
-  extends: [
-    'standard',
-    'prettier',
-    'plugin:import/recommended',
-    'plugin:jest-formatting/recommended',
-    'plugin:n/recommended',
-    'plugin:promise/recommended'
-  ],
+  ignorePatterns: ['.server', '.public', 'src/__fixtures__', 'coverage'],
   overrides: [
     {
+      extends: [
+        'standard',
+        'plugin:import/recommended',
+        'plugin:import/typescript',
+        'plugin:jsdoc/recommended-typescript-flavor',
+        'plugin:n/recommended',
+        'plugin:promise/recommended',
+        'plugin:@typescript-eslint/recommended-type-checked',
+        'plugin:@typescript-eslint/stylistic-type-checked',
+        'prettier'
+      ],
+      env: {
+        browser: false
+      },
+      files: ['**/*.{cjs,js}'],
+      parser: '@typescript-eslint/parser',
+      parserOptions: {
+        ecmaVersion: 'latest',
+        project: ['./tsconfig.json'],
+        tsconfigRootDir: __dirname
+      },
+      plugins: [
+        '@typescript-eslint',
+        'import',
+        'jsdoc',
+        'n',
+        'promise',
+        'prettier'
+      ],
+      rules: {
+        'prettier/prettier': [
+          'error',
+          {
+            endOfLine: 'auto'
+          }
+        ],
+        'no-console': 'error',
+
```

Comment:
Also worth noting. These are currently correct and have turned off the type checking rules as described in - [
Turn off strict type checking rules + config](https://github.com/DEFRA/cdp-node-frontend-template/commit/2e1ebbfa7ce903f160f9acacce885e5489562d51)

To double check. Have you turned these on and off and ran type checks? To see if there are any errors in this repo that need fixing. Before we then turn it off by default in the repo?

Reply:
There's some issues getting the Hapi server types working nicely with decorators (and presumably pre helpers too) since they dynamically extend the base class.